src/tools/rustfmt/src/items.rs

```diff
@@ -985,7 +988,7 @@ fn format_impl_ref_and_type(
 
     let trait_ref_overhead;
     if let Some(of_trait) = of_trait.as_deref() {
-        result.push_str(format_constness_right(of_trait.constness));
+        result.push_str(format_constness_right(*constness));
```

Comment:
are we not going to format `impl const Tr` to `const impl Tr`? Feels like it should be placed at the earlier places now.